### PR TITLE
Feat/cstm cpc 878 add proper persistency to the pet details page

### PR DESCRIPTION
**Pull request by DisgruntledMensch**
JIRA: [link to jira ticket
](https://champlainsaintlambert.atlassian.net/browse/CPC-878)## Context:
What is the ticket about and why are we doing this change.
Added proper methods for grabbing a pets status and updating the ui to reflect changes from a status patch or pet delete

## Before and After UI (Required for UI-impacting PRs)
**On page load**
image
**Clicking the toggle**
image
**Delete**
image

### CODE REVIEW

File: api-gateway/src/main/resources/static/scripts/owner-details/owner-details.controller.js [hunk 1]
```diff
@@ -90,9 +96,11 @@ function OwnerDetailsController($http, $state, $stateParams, $scope, $timeout, $
             console.error('Error fetching pet data:', error);
         });
 
+
+
     // Toggle pet's active status
     vm.toggleActiveStatus = function (petId) {
-        $http.get('api/gateway/pets/' + petId + '?_=' + new Date().getTime(), { headers: { 'Cache-Control': 'no-cache' } })
+        return $http.get('api/gateway/pets/' + petId + '?_=' + new Date().getTime(), { headers: { 'Cache-Control': 'no-cache' } })
```
Comment (Nmazza1) [inline api-gateway/src/main/resources/static/scripts/owner-details/owner-details.controller.js:103]: Not sure i understand the query here

**Review by Nmazza1 — DISMISSED** [review]
Looks good overall, was the data persistency fixed by my fix or your caching?

**Review by 1-emil — DISMISSED** [review]
The code looks great I do not see any issues with it.

**Review by Nmazza1 — APPROVED** [review]
Classic Out of date branch moment

**Review by 1-emil — APPROVED** [review]
Code looks great I don't see any issues

**Pull request merged**